### DISCUSSION TITLE
How to update only specific rows in v7

### DISCUSSION THREAD

**Discussion by thefaded** [TanStack/table#2340] [General] [answered]
I'm using Redux, and for table I'm compute derived data using reselect, after this assign it from props to state.
```
const [data, setData] = useState(() => props.rowData)
const columns         = useMemo(generatedColumns, [props.projectionsMonthsNumber])

// Update rowData externally
useEffect(() => setData(props.rowData), [props.rowData])
```

In the 2nd column I have input, so on each onChange I should recalculate data for just one row, I'm doing it this way
```
      Cell: ({
        row: { index },
        column: { id },
        cell: { row: { values: { sales, sku } } },
        updateMyData
      }) => {
        const [value, setValue] = React.useState(sales)
  
        // If the initialValue is changed external, sync it up with our state
        useEffect(() => {
          setValue(sales)
        }, [sales])
    
        const onChange = e => {
          setValue(e.target.value)
          updateMyData(index, id, e.target.value)
        }

        return <input value={value} onChange={onChange} />
      }
```
And here I'm generating new table's body
```
      <Table.Body {...getTableBodyProps()}>
        {rows.map((row, i) => {
          prepareRow(row)
          return bodyRowRenderer(row)
        })}
      </Table.Body>
```

But the problem is what after updating only single row I have the whole table re-rendering and it's pretty slow, how can avoid it? So just single row must be re-rendered

**Comment by dbertella** [comment]
Did you check this section in FAQ? https://github.com/tannerlinsley/react-table/blob/master/docs/faq.md#how-do-i-stop-my-table-state-from-automatically-resetting-when-my-data-changes

**Reply by robgwin** [reply]
Every post I can find about this issue leads to that FAQ link, which is a 404. Any clue where it went?

**Reply by cavvon** [reply]
It's now here https://react-table.tanstack.com/docs/faq#how-do-i-stop-my-table-state-from-automatically-resetting-when-my-data-changes

**Reply by divmgl** [reply]
Lol It's 404ing again.

**Reply by leahlang4d** [reply]
It's here now https://react-table-v7.tanstack.com/docs/faq#how-do-i-stop-my-table-state-from-automatically-resetting-when-my-data-changes

**Reply by muroa96** [reply]
I think [docs/faq.md](https://github.com/TanStack/table/blob/main/docs/faq.md) in the repo might be same with above link.

**Comment by thefaded** [comment, marked as answer]
Generally, I would say only `React.memo` with custom `equalityCheck` function works, nothing else. I found solution here – https://github.com/tannerlinsley/react-table/issues/1496

But the main problem I think is that there are a lot of useless posts/comments/articles about that React is fast and you don't have to optimise anything in your components, as we can see reference to Kent C. Dodds article in the link above. If I'm saying it's slow, it mens it's slow and don't say that it's fast.
Why my table with 2k cells must be re-rendered if only single row is changed? I should go and optimise render function/component? Hell no, react must re-render only single row/cells what really updated, not everything, it's React job.
With re-rendering whole table we can just go on and omit using react and just use `CMR + R`, cool solution 😎 ?

It would be cool if you update docs, covering this topic, because I pretty often see discussions on this problem

**Reply by tannerlinsley** [reply]
What if that single change results in filters, grouping or sorting becoming out of sync?

**Reply by arichiardi** [reply]
FWIW relying on row index would not work in case of grouping.

**Reply by User6531** [reply]
@thefaded Hey there, could you please share a solution for this? I'm dealing with the same issue of having to rerender the entire table when just one row gets updated.

Do you think using React.memo with td might help in this case?

**Comment by AshutoshSajan** [comment]
I am having the same issue, I have cell input and whenever I am updating a single cell it re-renders the whole table

**Comment by boldwade** [comment]
I have the same problem on v8... 
I'm trying to understand, what's the purpose of this library at all if I have to manually handle everything myself regarding updating data in place? Just seems like there are plenty of other libraries out there that handle this better?

**Reply by dcmwong** [reply]
Can you name some better libraries please.
